### PR TITLE
Additional Test Cases for parser

### PR DESCRIPTION
**Pull request by MoniMoledo**
Additional test cases for JSON Parser to increase coverage

@JavierJia

### CODE REVIEW

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala [hunk 1]
```diff
@@ -34,5 +35,36 @@ class JSONParserTest extends Specification with TestQuery {
       val expectQuery = Query(schema.dataset, Seq.empty, filter, Seq.empty, None, Some(selectRecent))
       actualQuery must_== expectQuery
     }
+    "parse long values " in {
```
Comment (JavierJia) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: nice, could we add all types?  `parse int `, `parse string`.
I haven't think about if `bag` is applicable for values, do you think it is reasonable to pass a array of `bag`s? if not we should throw warnings.

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala [hunk 2]
```diff
@@ -34,5 +35,36 @@ class JSONParserTest extends Specification with TestQuery {
...
+      val expectQuery = new Query(schema.dataset, Seq.empty, Seq(doubleFilter), Seq.empty, None, None)
+      actualQuery must_== expectQuery
+    }
+    "parse geo-cell group function " in {
+     val actualQuery = parser.parse(geoCellJSON)
+      val expectQuery = new Query(schema.dataset, Seq.empty, Seq.empty, Seq.empty,
+        Some(GroupStatement(Seq(byGeocell), Seq(aggrCount))), None)
+      actualQuery must_== expectQuery
+    }
+    "parse is_retweet filter request" in {
```
Comment (JavierJia) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: would it be `parse boolean filter`?

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala [hunk 3]
```diff
@@ -34,5 +35,36 @@ class JSONParserTest extends Specification with TestQuery {
...
+        Some(GroupStatement(Seq(byUser), Seq(aggrCount))), None)
+      actualQuery must_== expectQuery
+    }
+    "throw an exception when there is no dataset name" in {
+      parser.parse(missingDatasetJSON) must throwA[JsonRequestException]
+    }
+    "throw an exception when value is an array" in {
+      parser.parse(filterErrorJSON) must throwA[JsonRequestException]
+    }
+    "throw an exception when relation is unknown" in {
```
Comment (JavierJia) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: nice

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala [hunk 4]
```diff
@@ -14,17 +14,28 @@ trait TestQuery {
   val timeFilter = FilterStatement("create_at", None, Relation.inRange, Seq(startTime, endTime))
   val textFilter = FilterStatement("text", None, Relation.contains, textValue)
   val stateFilter = FilterStatement("geo_tag.stateID", None, Relation.in, stateValue)
+  val retweetFilter = FilterStatement("is_retweet", None, Relation.isTrue, Seq.empty)
+
+  val longValues: Seq[Long] = Seq(1644.toLong, 45464.toLong)
```
Comment (JavierJia) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala]: I think you can just write `1644l, 45464l`.

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala [hunk 5]
```diff
@@ -51,6 +62,62 @@ trait TestQuery {
...
+     """.stripMargin
+
+
+  val filterRetweetJSON =
+    s"""
+       |"filter": [
+       |  {
+       |    "field": "is_retweet",
+       |    "relation": "true",
+       |    "values": []
```
Comment (JavierJia) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala]: do you want to make the `values` optional

**Comment by JavierJia** [comment]
looks good. just left some minor comments.

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala (hunk 2 above)
Comment (MoniMoledo) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: Yes, better! Done

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala (hunk 4 above)
Comment (MoniMoledo) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala]: Thanks, I was trying to find something like this. Done

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala (hunk 5 above)
Comment (MoniMoledo) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala]: Currently this is the only case "values" is not required. I can change it to optional for now. But I think a better approach would be later change the boolean relation to "is" and values:[true/false]

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala (hunk 1 above)
Comment (MoniMoledo) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: Yes for the types. 
About the bags, tricky question. I can't think in a usage for bags as values right now.

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala (hunk 3 above)
Comment (MoniMoledo) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: thnks!

**Comment by JavierJia** [comment]
Looks good. 👍

**Pull request merged**